src/sparkle/writer/kafka_writer.py

```diff
@@ -109,3 +109,91 @@ def write(self, df: DataFrame) -> None:
             .option("topic", self.kafka_topic)
             .start()
         )
+
+
+class KafkaBatchPublisher(Writer):
+    """KafkaBatchublisher class for writing DataFrames in batch to Kafka.
+
+    Inherits from the Writer abstract base class and implements the write
+    method for writing data to Kafka topics.
+
+    Args:
+        kafka_options (dict[str, Any]): Kafka connection options.
+        kafka_topic (str): Kafka topic to which data will be written.
+        unique_identifier_column_name (str): Column name used as the Kafka key.
+        spark (SparkSession): Spark session instance to use.
+    """
+
+    def __init__(
+        self,
+        kafka_options: dict[str, Any],
+        kafka_topic: str,
+        unique_identifier_column_name: str,
+        spark: SparkSession,
+    ) -> None:
+        """Initialize the KafkaBatchPublisher object.
+
+        Args:
+            kafka_options (dict[str, Any]): Kafka options for the connection.
```

Comment:
With regards to your communication in Slack @wtfzambo I completely agree with you, this kind of configuration, is contradicting the whole purpose of `with_config`. So initially when we introduced it in the previous iteration, was we were taking in the kafka_options, as an argument list in the construct. If that's not necessary, why not just take in the KafkaConfig type here as the input to initialize the construct and drop `with_config` altogether?

Reply:
Admittedly, I still have limited vision, but I'd also drop the `with_config` method.

Especially if `KafkaReader` and `KafkaWriter` are supposed to be made public to the user.

The only counter-aspect that I can think of is that passing the whole `Config` object is a lot more verbose than the current constructor method, which takes just the bare minimum of values to work with Kafka.